accounts/forms.py

```diff
@@ -312,8 +312,15 @@ class DeleteUserForm(forms.Form):
     encrypted_link = forms.CharField(widget=forms.HiddenInput())
     delete_sounds = forms.ChoiceField(choices = DELETE_CHOICES,
             widget=forms.RadioSelect())
+    password = forms.CharField(label=_("Your password"), widget=forms.PasswordInput)
 
-    def clean_encrypted_link(self):
+    def clean_password(self):
+        user = User.objects.get(id=self.user_id)
+        if not user.check_password(self.cleaned_data["password"]):
+            raise forms.ValidationError(_("Incorrect password."))
+        return self.cleaned_data['password']
```

Comment:
I guess you should return `None` or `""` here not to have the password be played back to the user? If the user messed up the password is still going to be repeated to the user from the backend... That's a bit messy.